### PR TITLE
Add release-1.10 Prow job configs

### PR DESCRIPTION
**Pull request by Monkeyanator**
Part of [https://github.com/istio/istio/issues/31627](https://github.com/istio/istio/issues/31627)

### CODE REVIEW

File: prow/config/jobs/common-files-1.10.yaml [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  - cache
+  - github
+  types:
+  - postsubmit
+- command:
+  - ../test-infra/tools/automator/automator.sh
+  - --org=istio
+  - --repo=test-infra
+  - '--title=Automator: update build-tools:$AUTOMATOR_SRC_BRANCH'
+  - --branch=master
```
Comment (howardjohn) [inline prow/config/jobs/common-files-1.10.yaml:58]: is this right?

File: prow/config/jobs/proxy-1.10.yaml [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    testing: build-pool
+  requirements:
+  - cache
+  - gcp
+  timeout: 4h0m0s
+  types:
+  - presubmit
+- command:
+  - ./prow/proxy-presubmit-centos-release.sh
+  image: gcr.io/istio-testing/build-tools-centos:master-2021-04-02T23-15-48
```
Comment (howardjohn) [inline prow/config/jobs/proxy-1.10.yaml]: this looks wrong

File: prow/config/jobs/proxy-1.10.yaml [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+- command:
+  - ../test-infra/tools/automator/automator.sh
+  - --org=istio
+  - --repo=istio
+  - '--title=Automator: update proxy@$AUTOMATOR_SRC_BRANCH in $AUTOMATOR_ORG/$AUTOMATOR_REPO@$AUTOMATOR_BRANCH'
+  - --labels=auto-merge,release-notes-none
+  - --modifier=update_proxy_dep
+  - --token-path=/etc/github-token/oauth
+  - --cmd=bin/update_proxy.sh $AUTOMATOR_SHA
+  image: gcr.io/istio-testing/build-tools:master-2021-04-02T23-15-48
```
Comment (howardjohn) [inline prow/config/jobs/proxy-1.10.yaml]: this seems wrong?

**Review by howardjohn — COMMENTED** [review]
I think there are still some suspicious master usages?

File: prow/config/jobs/common-files-1.10.yaml (hunk 1 above)
Comment (ericvn) [reply prow/config/jobs/common-files-1.10.yaml:58]: Need to investigate. It is `master` in the other release yamls.

File: prow/config/jobs/proxy-1.10.yaml (hunk 2 above)
Comment (ericvn) [reply prow/config/jobs/proxy-1.10.yaml]: Agree. Looks like generate.go only changes the outer image, and not the inner images, or there is something changing it to master. Adding a note to investigate the generate.go and will fix manually.

File: prow/config/jobs/proxy-1.10.yaml (hunk 3 above)
Comment (Monkeyanator) [reply prow/config/jobs/proxy-1.10.yaml]: Changed all build-tools references to release-1.10

File: prow/config/jobs/common-files-1.10.yaml (hunk 1 above)
Comment (ericvn) [reply prow/config/jobs/common-files-1.10.yaml:58]: Reason this is master is that the repo is test-infra and that only has `master`.

**Review by ericvn — APPROVED** [review]

**Pull request merged**